pkg/service/publish_logs.go

```diff
@@ -102,6 +102,9 @@ func encodeGRPCLogCollection(_ context.Context, request interface{}) (interface{
 func decodeGRPCPublishLogsResponse(_ context.Context, grpcReq interface{}) (interface{}, error) {
 	req := grpcReq.(*pb.AgentApiResponse)
 	return publishLogsResponse{
+		jsonRpcResponse: jsonRpcResponse{
+			DisableDevice: req.DisableDevice,
+		},
 		Message:     req.Message,
```

Comment:
Naive question -- why are the other things really obvious looking struct members, but this one is a nested thing?

Reply:
this is because the jsonRpc implementation is doing a standard json unmarshalling where as the grpc implementation is taking the auto-generated protobuf struct`pb.AgentApiResponse` and converting it the struct type `publishLogsResponse` via manually setting the fields

we could also probably just marshall the protobuf to json then unmarshall it into the `publishLogsResponse`

Reply:
Hrm. Caveat, I'm not sure what correct is...

My hunch is that we should do the same thing for jsonrpc and grpc, but I'm not sure if we should improve the conversion or just go the manual route. 

At some point, I think we should level up all the API surfaces via some kind of API specification, but that feels a bit out of scope for this.